tests/validation/tests/v3_api/test_bkp_restore_s3_recover.py

```diff
@@ -0,0 +1,169 @@
+import pytest
+from .common import *  # NOQA
+from .test_rke_cluster_provisioning import rke_config, \
+    create_and_validate_custom_host
+
+namespace = {"p_client": None, "ns": None, "cluster": None, "project": None,
+             "nodes": []}
+backup_info = {"backupname": None, "backup_id": None, "workload": None,
+               "backupfilename": None, "etcdbackupdata": None}
+HOST_NAME = os.environ.get('RANCHER_HOST_NAME', "testcustom")
+
+rke_config["services"]["etcd"]["backupConfig"] = {
+    "enabled": "true",
+    "intervalHours": 12,
+    "retention": 6,
+    "type": "backupConfig",
+    "s3BackupConfig": {
+        "type": "s3BackupConfig",
+        "accessKey": AWS_ACCESS_KEY_ID,
+        "secretKey": AWS_SECRET_ACCESS_KEY,
+        "bucketName": AWS_S3_BUCKET_NAME,
+        "folder": AWS_S3_BUCKET_FOLDER_NAME,
+        "region": AWS_REGION,
+        "endpoint": "s3.amazonaws.com"
+    }
+}
+
+
+@if_test_all_snapshot
+def test_bkp_restore_s3_recover_validate():
+    cluster = namespace["cluster"]
+    client = namespace["client"]
+    ns, b_info = validate_backup_create(namespace, backup_info, "s3")
+    etcd_nodes = get_etcd_nodes(client, cluster)
+    assert len(etcd_nodes) > 0, "Make sure we have etcd nodes in the cluster"
+
+    # Proceed to remove all etcd nodes
+    ips_to_remove = []
+    for etcd_node in etcd_nodes:
+        ips_to_remove.append(etcd_node.externalIpAddress)
+        client.delete(etcd_node)
+    cluster = wait_until_cluster_is_updating(client, cluster)
+    # Also remove the ec2 instances
+    for ip_to_remove in ips_to_remove:
+        delete_node_from_ec2(ip_to_remove)
+    ips_to_remove.clear()
+    cluster = client.reload(cluster)
+    etcd_nodes = get_etcd_nodes(client, cluster)
+    assert len(etcd_nodes) == 0, "Ensure the cluster has zero etcd nodes now"
+    # Add completely new etcd nodes to the cluster
+    cluster = add_new_etcd_nodes(client, cluster)
+    cluster = client.reload(cluster)
+    # This message is expected to appear after we add new etcd nodes
+    # The cluster will require the user to perform a backup to recover
+    # this is appears in the cluster object in cluster.transitioningMessage
+    message = "Please restore your cluster from backup"
+    cluster = wait_for_cluster_transitioning_message(client, cluster, message)
+    etcd_nodes = get_etcd_nodes(client, cluster)
+    assert len(etcd_nodes) == 3, "Make sure the cluster now has 3 etcd nodes"
+    restore_backup_to_recover(client, cluster, b_info)
```

Comment:
`restore_backup_to_recover` should return cluster object ?

Reply:
Not of much use at that stage of the scenario but I will store the returned value.